JustSaying/JustSayingFluently.cs

```diff
@@ -406,5 +406,11 @@ public IMayWantOptionalSettings WithAwsClientFactory(Func<IAwsClientFactory> aws
             _awsClientFactoryProxy.SetAwsClientFactory(awsClientFactory);
             return this;
         }
+
+        public void Dispose()

```

Comment:
If the class is public and not sealed, it should implement with the `protected void Dispose(bool disposing)` pattern with a finalizer so that derived types can participate in disposal too.

Reply:
The dispose pattern has been an anti-pattern for a number of years now, so I'd suggest leaving it. I'll sends some links tomorrow to clarify.

Reply:
What I was trying to say (badly) was "seal this class".

Reply:
When inheritance is expected and you don't use the pattern, integrators have a bad time when they want to dispose things they add in a derived types, leading to me having to raise PRs like this as otherwise you end up with _horrible_ code using `new` to replace implementations that cause whole heaps of sadness: aspnet/Mvc#7637

Reply:
Agreed, in the example you sent that could have been avoided if they had made the dispose method virtual. To be fair, the dispose pattern makes sense for Microsoft as they are building the building blocks for all types of crazy that people might do, but I really think they should change the guidance for everyone else.

Agreed with "seal this class" 👍